### PR TITLE
Allow launcher in nodejs

### PR DESCRIPTION
**Pull request by martaver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/bazelbuild/rules_docker/issues/1802

"entrypoint / launcher in nodejs_image isn't respected "

## What is the new behavior?

The `launcher` and `launcher_args` attributes are now accepted by `nodejs_image` and passed to `app_layer`.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This feature isn't really node specific. If maintainers agree, then I would update all the language images that use `app_layer` to accept `launcher` and `launcher_args`. Then I would also remove the documentation change from `nodejs_image` and instead make a general note that all the language images support the attributes.

### CODE REVIEW

File: run-bazel-in-docker.sh
Comment (alexeagle) [inline run-bazel-in-docker.sh:1]: did you mean to check in this file? I guess it's for testing something. If so, maybe add some comments here about what it's used for and how to use it

**Review by alexeagle — COMMENTED** [review]
This is awesome thanks!

You asked on slack about whether this should cover other languages, I don't really know...

Before we merge this, I have one question though. Would it be hard for a user to just skip using the convenient language rules like `nodejs_image` and just make their own `container_image` in order to get access to these attributes? maybe the use case is narrow enough that omitting them from the easy-to-use language wrappers is actually a feature to make things simpler for common users.

File: run-bazel-in-docker.sh
Comment (martaver) [reply run-bazel-in-docker.sh]: Added some docs... I created this script because the repo's tests fail on non-linux systems (I'm on MacOS):
```
tests/container/nodejs/nodejs_image_custom_binary_test.image: line 208: /usr/bin/docker: No such file or directory
```

**Comment by martaver** [comment]
re: rolling your own `container_image`... I'd rather avoid that... `nodejs_image` forms a standard that I want to be able to lean on wherever possible. Deviating from it should be a last resort.

I would argue that launchers are common (roughly 50% of the images I deal with have them) and a minor enough implementation detail that it shouldn't warrant having to roll a custom `container_image`.

At the end of the day, there are basically only two ways to customise a `container_image`... change the base image, and the launch script / entry point. `nodejs_image` supports the former already, but not the latter.

This is also why I suspect it'd be good form to adder `launcher` and `launcher_args` to all the lang images.

**Comment by alexeagle** [comment]
I tried to rebase this for you to land it, but I think you disabled the checkbox "allow edits from maintainers" - could you rebase yourself or check the box?

**Review by alexeagle — APPROVED** [review]

**Comment by martaver** [comment]
Mmmmm done... looks like `.gitignore` had a conflict...

What did you think of exposing the launcher args for each language image btw?

**Review by alexeagle — APPROVED** [review]
If the same problem exists for other languages, then yeah we should fix it for all of them.

I'm okay with getting this fix in first, I don't think it's practical to make you do all of them at once.

**Pull request merged**
